scripts/check_no_module_imports.sh

```diff
@@ -0,0 +1,30 @@
+# Copyright 2018 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Fail if any source files contain Objective-C module @imports, excluding:
+#   * Example sources
+#   * Sample sources
+#
+options=(
+  -n  # show line numbers
+  -I  # exclude binary files
+  '^@import'
+)
+
+git grep "${options[@]}" \
+    -- ':(exclude,glob)**/Example/**' ':(exclude,glob)**/Sample/**'
```

Comment:
This excludes unit tests as well as Example code.   The tests are in

```
 $ ls -ld Example/*/Tests */Example/Tests
drwxr-xr-x  60 paulbeusterien  eng  1920 Mar 28 13:18 Example/Auth/Tests
drwxr-xr-x  11 paulbeusterien  eng   352 Mar 28 12:15 Example/Core/Tests
drwxr-xr-x   9 paulbeusterien  eng   288 Mar 28 12:16 Example/Database/Tests
drwxr-xr-x  24 paulbeusterien  eng   768 Mar 28 12:16 Example/Messaging/Tests
drwxr-xr-x   5 paulbeusterien  eng   160 Mar 28 12:16 Example/Storage/Tests
drwxr-xr-x  13 paulbeusterien  eng   416 Mar 28 12:16 Firestore/Example/Tests
drwxr-xr-x   6 paulbeusterien  eng   192 Mar 28 12:16 Functions/Example/Tests
```

Reply:
Ah, I didn't understand these were the main tests for Firebase.

Updated all test sources files to change `@import` to `#import`.

Updated dependent frameworks in the Xcode project framework link step for the tests.